File: app/src/main/java/com/hustunique/morii/database/DiaryWithSoundItemInfo.java
package com.hustunique.morii.database;

import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Relation;

import java.util.List;

public class DiaryWithSoundItemInfo {
    @Embedded
    public DiaryInfo diaryInfo;

    @Relation(
            parentColumn = "id",
            entityColumn = "diaryInfoId"
    )
    public List<SoundItemInfo> soundItemInfoList;
}

File: app/src/main/java/com/hustunique/morii/database/DiaryWithSoundItemInfo.kt
package com.hustunique.morii.database

import androidx.room.*

data class DiaryWithSoundItemInfo(
    @Embedded
    val diaryInfo: DiaryInfo,

    @Relation(parentColumn = "id", entityColumn = "diaryInfoId")
    val soundItemInfoList: List<SoundItemInfo>
)